lib/connect.lua

```diff
@@ -148,7 +150,11 @@ local function connect(mapStateToPropsOrThunk, mapDispatchToProps)
 				propsForChild = nil,
 			}
 
-			self.state.propsForChild = stateUpdater(self.props, self.state, mappedStoreState)
+			local extraState = stateUpdater(self.props, self.state, mappedStoreState)
+
+			for key, value in pairs(extraState) do
+				self.state[key] = value
```

Comment:
I'm not sure what this is doing, but I'm fairly certain it's not self.state that we want to be updating here.